### PR TITLE
Rebase of PR #7556

### PR DESCRIPTION
**Pull request by rustyrussell**
I think this:

Closes: #7575
Closes: #7556 ?

### CODE REVIEW

**Review by cdecker — APPROVED** [review]
ACK 73e83c6f3699fb413130b590c0ec52b09734f028

**Comment by cdecker** [comment]
This might require an update mocks call, the flake build is failing in run route overlong not having the mocks for tracing related functions.

File: plugins/libplugin.c [hunk 1]
```diff
@@ -1058,7 +1059,14 @@ static void handle_rpc_reply(struct plugin *plugin, const jsmntok_t *toks)
 	cmd_freed = false;
 	tal_add_destructor2(out->cmd, destroy_cmd_mark_freed, &cmd_freed);
 
+	trace_span_resume(out);
 	contenttok = json_get_member(buf, toks, "error");
+
+	/* Annotate the JSON-RPC span whether it succeeded or failed,
+	 * and then emit it. */
+	trace_span_tag(out, "error", contenttok == NULL ? "true" : "false");
```
Comment (JssDWt) [inline plugins/libplugin.c:1067]: This needs to be the other way around

```suggestion
	trace_span_tag(out, "error", contenttok == NULL ? "false" : "true");
```

**Review by JssDWt — COMMENTED** [review]
One nit, that I thought I had fixed in #7556, but apparently not.

File: plugins/libplugin.c (hunk 1 above)
Comment (rustyrussell) [reply plugins/libplugin.c:1067]: That's two negatives.  I will change it  to `contenttok ? "true" : "false"`

**Review by JssDWt — APPROVED** [review]
ACK b65fcbe13237493341a8b9a6363179d8aa4ddd48

**Comment by vincenzopalazzo** [comment]
Unrelated faulure inside the CI 

```
lightningd-1 2024-11-13T07:15:15.499Z DEBUG   gossipd: Shutting down
lightningd-1 2024-11-13T07:15:15.532Z DEBUG   hsmd: Shutting down
{'github_repository': 'ElementsProject/lightning', 'github_sha': '254d079e69501119aae75f7ea200d5bc43c7c6da', 'github_ref': 'refs/pull/7803/merge', 'github_ref_name': 'HEAD', 'github_run_id': 11811119714, 'github_head_ref': 'guilt/pr-7556', 'github_run_number': 11470, 'github_base_ref': 'master', 'github_run_attempt': '2', 'testname': 'test_important_plugin', 'start_time': 1731482015, 'end_time': 1731482115, 'outcome': 'success'}
----------------------------- Captured stderr call -----------------------------
Lost connection to the RPC socket.Reading JSON input: Connection reset by peerReading JSON input: Connection reset by peerReading JSON input: Connection reset by peer
--------------------------- Captured stdout teardown ---------------------------
['lightningd-1 2024-11-13T07:14:12.293Z **BROKEN** plugin-spenderp: Lost connection to the RPC socket.']
===Flaky Test Report===

```

merging this, and opening an issue for the CI faulure

**Pull request merged**